lib/alchemy/pg_search/search.rb

```diff
@@ -0,0 +1,59 @@
+module Alchemy
+  module PgSearch
+    module Search
+
+      ##
+      # index all supported Alchemy models
+      def self.rebuild
+        ([Alchemy::Page, Alchemy::Ingredient] + Alchemy::PgSearch.searchable_essence_classes).each do |model|
+          ::PgSearch::Multisearch.rebuild(model)
+        end
+      end
+
+      ##
+      # remove the whole index for the page
+      #
+      # @param page [Alchemy::Page]
+      def self.remove_page(page)
+        ::PgSearch::Document.delete_by(page_id: page.id)
+      end
+
+      ##
+      # index a single page and indexable essences
+      #
+      # @param page [Alchemy::Page]
+      def self.index_page(page)
+        remove_page page
+
+        page.update_pg_search_document
+        page.all_elements.includes(contents: :essence).each do |element|
```

Comment:
Eventuell wollen noch eine Optimierung für sehr große elements table machen

```suggestion
        page.all_elements.includes(contents: :essence).find_each do |element|
```

Das lädt die Elements in packen zu tausend anstatt alle auf einmal

Reply:
Und wir wollen auch die `ingredients` eager loaden.

```suggestion
        page.all_elements.includes(contents: :essence, :ingredients).find_each do |element|
```